helm/dagster/templates/configmap-instance.yaml

```diff
@@ -116,3 +116,15 @@ data:
 
     telemetry:
       enabled: {{ .Values.telemetry.enabled }}
+
+    {{- if .Values.retention }}
+    retention:
+      {{- if .Values.retention.sensor }}
+      sensor:
+        purge_after_days: {{ .Values.retention.sensor.purgeAfterDays | toYaml | nindent 12 }}
+      {{- end }}
+      {{- if .Values.retention.schedule }}
+      schedule:
+        purge_after_days: {{ .Values.retention.schedule.purgeAfterDays | toYaml | nindent 12 }}
+      {{- end }}
+    {{- end }}
```

Comment:
for backcompat, should include a `.Values.retention.enabled` field. As it stands, this field will always be included.